server/game/GameActions/LastingEffectCardAction.ts

```diff
@@ -5,6 +5,8 @@ import BaseCard = require('../basecard');
 import { LastingEffectGeneralProperties } from './LastingEffectAction';
 
 export interface LastingEffectCardProperties extends LastingEffectGeneralProperties {
+    match?: (card: BaseCard, context?: AbilityContext) => boolean;
```

Comment:
Neither of these properties should be necessary - the match is always the target of the game action, and match should always be a card, not a function for a lasting effect, so targetController is unnecessary